tests/HydratorTest.php

```diff
@@ -0,0 +1,659 @@
+use Yiisoft\Hydrator\Tests\Support\Attribute\FromPredefinedArrayResolver;
+use Yiisoft\Hydrator\Tests\Support\Attribute\InvalidParameterResolver;
+use Yiisoft\Hydrator\Tests\Support\Attribute\NotResolver;
+use Yiisoft\Hydrator\Tests\Support\Model\ConstructorParameterAttributesModel;
+use Yiisoft\Hydrator\Tests\Support\Model\ConstructorTypeModel;
+use Yiisoft\Hydrator\Tests\Support\Model\CounterModel;
+use Yiisoft\Hydrator\Tests\Support\Model\FromPredefinedArrayModel;
+use Yiisoft\Hydrator\Tests\Support\Model\InvalidDataResolverModel;
+use Yiisoft\Hydrator\Tests\Support\Model\ObjectPropertyModel\ObjectPropertyModel;
+use Yiisoft\Hydrator\Tests\Support\Model\ObjectPropertyModel\RedCar;
+use Yiisoft\Hydrator\Tests\Support\Model\StaticModel;
+use Yiisoft\Hydrator\Tests\Support\Model\TypeModel;
+use Yiisoft\Hydrator\Tests\Support\Model\NestedModel\UserModel;
+use Yiisoft\Hydrator\Tests\Support\Model\PreparePropertyModel;
+use Yiisoft\Hydrator\Tests\Support\Model\SimpleModel;
+use Yiisoft\Hydrator\Tests\Support\String42Typecast;
+use Yiisoft\Hydrator\Tests\Support\StringableObject;
+use Yiisoft\Hydrator\Typecast\CompositeTypecast;
+use Yiisoft\Hydrator\Typecast\NoTypecast;
+use Yiisoft\Hydrator\Typecast\SimpleTypecast;
+use Yiisoft\Test\Support\Container\SimpleContainer;
+
+final class HydratorTest extends TestCase
+{
+    public function testSimpleCreate(): void
+    {
+        $service = new Hydrator(new SimpleContainer());
+
+        $model = $service->create(
+            SimpleModel::class,
+            ['a' => '1', 'b' => '2', 'c' => '3'],
+        );
+
+        $this->assertSame('1', $model->getA());
+        $this->assertSame('2', $model->getB());
+        $this->assertSame('3', $model->getC());
+    }
+
+    public function testSimpleHydrate(): void
+    {
+        $service = new Hydrator(new SimpleContainer());
+
+        $model = new SimpleModel();
+        $service->hydrate(
+            $model,
+            ['a' => '1', 'b' => '2', 'c' => '3'],
+        );
+
+        $this->assertSame('1', $model->getA());
+        $this->assertSame('2', $model->getB());
+        $this->assertSame('3', $model->getC());
+    }
+
+    public function testSimpleCreateStrict(): void
+    {
+        $service = new Hydrator(new SimpleContainer());
+
+        $model = $service->create(
+            SimpleModel::class,
+            ['a' => '1', 'b' => '2', 'c' => '3'],
+            ['b' => 'b'],
+            true,
+        );
+
+        $this->assertSame('.', $model->getA());
+        $this->assertSame('2', $model->getB());
+        $this->assertSame('.', $model->getC());
+    }
+
+    public function testSimpleHydrateStrict(): void
+    {
+        $service = new Hydrator(new SimpleContainer());
+
+        $model = new SimpleModel();
+        $service->hydrate(
+            $model,
+            ['a' => '1', 'b' => '2', 'c' => '3'],
+            ['b' => 'b'],
+            true,
+        );
+
+        $this->assertSame('.', $model->getA());
```

Comment:
Why so?

Reply:
Strict mode. In strcit mode hydrate properties from map only.

Reply:
I mean why the dot, but not an empty string, null or exception?

Reply:
In strict mode not hydrate properties which are not in map. In this test dots is default values in model:

![image](https://user-images.githubusercontent.com/525501/236659335-8558d638-4f01-4f3a-ab5a-ca68a33b69d8.png)

Reply:
And what would happen if a model doesn't have default value?

Reply:
Nothing